processor/k8sprocessor/kube/client.go

```diff
@@ -104,7 +115,13 @@ func (c *WatchClient) Start() {
 		UpdateFunc: c.handlePodUpdate,
 		DeleteFunc: c.handlePodDelete,
 	})
-	c.informer.Run(c.stopCh)
+	go c.informer.Run(c.stopCh)
+	c.namespaceInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc:    c.handleNamespaceAdd,
+		UpdateFunc: c.handleNamespaceUpdate,
+		DeleteFunc: c.handleNamespaceDelete,
+	})
+	c.namespaceInformer.Run(c.stopCh)
```

Comment:
This feels a bit awkward. Why would we run one informer as a go routine but another as a regular function. What implications does it have?  At first glace it looks like this method (Start) was earlier waiting for `c.informer.Run` function call to finish but won't do it after this change. Not sure how it affects the behavior. We should make sure it doesn't change the behavior in any way or at least understand all implications and share them in the PR or as a comment.

In case this does not make any practical difference, can we run both informers as go routines for consistency in code?

Reply:
we can run both as goroutines for consistency.

Reply:
Are you sure that is safe? Should the processor wait for the Run function to return before it can tell the pipeline it is ready to receive data or is that not the case?